src/gui/updater/appimageupdater.cpp

```diff
@@ -161,14 +161,24 @@ void AppImageUpdater::versionInfoArrived(const UpdateInfo &info)
         return;
     }
 
-    auto dialog = new AppImageUpdateAvailableDialog(currentVersion, newVersion, ocApp()->gui()->settingsDialog());
+    QPointer<AppImageUpdateAvailableWidget> widget(new AppImageUpdateAvailableWidget(currentVersion, newVersion, ocApp()->gui()->settingsDialog()));
+    widget->setAttribute(Qt::WA_DeleteOnClose);
 
-    connect(dialog, &AppImageUpdateAvailableDialog::skipUpdateButtonClicked, this, [newVersion]() {
+    connect(widget, &AppImageUpdateAvailableWidget::skipUpdateButtonClicked, this, [newVersion, widget]() {
         qCInfo(lcUpdater) << "Update" << newVersion << "skipped by user";
         setPreviouslySkippedVersion(newVersion);
+        if (widget) {
```

Comment:
As the connect is on the widget it must still exist at this time.
Please also use deleteLater.